src/rgw/rgw_admin.cc

```diff
@@ -1794,19 +1830,38 @@ static void get_md_sync_status(list<string>& status)
     return;
   }
 
+  string local_period = sync_status.sync_info.period;
   map<int, RGWMetadataLogInfo> master_shards_info;
   string master_period;
+  RGWRealm master_realm;
+  string remote,url,access_key,secret_key,realm_id,realm_name;
 
-  ret = sync.read_master_log_shards_info(&master_period, &master_shards_info);
+  /* Set the master zonegroup as the remote */
+  RGWPeriod current_period(local_period);
+  ret = current_period.init(g_ceph_context, store);
   if (ret < 0) {
+    status.push_back(string("failed to fetch local sync status") + cpp_strerror(-ret));
+    return;
+  }
+
+  remote = current_period.get_master_zonegroup();
+  ret = do_realm_get(remote, url, access_key, secret_key, realm_id, realm_name, master_realm);
+  if (ret < 0){
+    status.push_back(string("failed to fetch realm info from master: ") + cpp_strerror(-ret));
+    return;
+  }
+  ret = sync.read_master_log_shards_info(&master_period, &master_shards_info);
```

Comment:
yeah earlier we were checking against the `current_period` against`master_period` from the above call, which is why I thought the md sync status was always wrong (since that is the first period to start sync from). Will modify to `get_current_period_id`.